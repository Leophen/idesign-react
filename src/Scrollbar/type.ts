export interface ScrollbarProps {
  /**
   * 内容
   */
  children?: React.ReactNode;
  /**
   * 滚动容器高度
   */
  height?: React.CSSProperties["height"];
  /**
   * 滚动容器宽度
   */
  width?: React.CSSProperties["width"];
  /**
   * 水平滚动触发事件
   */
  onScrollX?: (x: number) => void;
  /**
   * 垂直滚动触发事件
   */
  onScrollY?: (y: number) => void;
}
