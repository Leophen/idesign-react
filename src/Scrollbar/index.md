---
nav:
  title: 组件
  path: /components
group:
  title: 基础组件
  order: 1
---

# Scrollbar 滚动条（开发中）

用于替换浏览器原生滚动条。

待定:

```tsx
import React from 'react';
import { Scrollbar } from 'idesign-react';

const App = () => {
  return (
    <Scrollbar height={300}>
      <li>123</li>
      <li>123</li>
      <li>123</li>
      <li>123</li>
      <li>123</li>
      <li>123</li>
      <li>123</li>
      <li>123</li>
      <li>123</li>
      <li>123</li>
      <li>123</li>
      <li>123</li>
      <li>123</li>
      <li>123</li>
      <li>123</li>
      <li>123</li>
      <li>123</li>
      <li>123</li>
      <li>123</li>
      <li>123</li>
      <li>123</li>
      <li>123</li>
      <li>123</li>
      <li>123</li>
    </Scrollbar>
  );
};

export default App;
```
