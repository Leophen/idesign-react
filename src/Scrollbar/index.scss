.i-scrollbar {
  position: relative;
}

.i-scrollbar__wrap {
  overflow: auto;
  &::-webkit-scrollbar {
    display: none;
  }
}

.i-scrollbar__barX,
.i-scrollbar__barY {
  position: absolute;
  z-index: 1;
}

.i-scrollbar__barX {
  bottom: 2px;
  left: 2px;
  width: 100%;
  height: 6px;
}

.i-scrollbar__barY {
  top: 2px;
  right: 2px;
  width: 6px;
  height: 100%;
}

.i-scrollbar__thumbX,
.i-scrollbar__thumbY {
  position: relative;
  background-color: rgba(144, 147, 153, 0.3);
  border-radius: 4px;
  cursor: pointer;
  transition: background-color 0.2s ease-out;
  &:hover {
    background-color: rgba(144, 147, 153, 0.5);
  }
}

.i-scrollbar__thumbX {
  width: 100px;
  height: 100%;
  transform: translateX(0%);
}

.i-scrollbar__thumbY {
  width: 100%;
  height: 100px;
  transform: translateY(0%);
}
