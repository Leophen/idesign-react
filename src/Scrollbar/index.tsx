import React, { Children } from 'react';
import './index.scss';

interface ScrollbarProps {
  children?: React.ReactNode;
}

const Scrollbar = (props: ScrollbarProps) => {
  const { children } = props;

  return <div className="i-design-scrollbar">{children}</div>;
};

export default Scrollbar;
