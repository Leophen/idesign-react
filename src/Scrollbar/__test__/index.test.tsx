import { render, fireEvent } from '@testing-library/react';
import Scrollbar from '../index';

describe('Scrollbar 组件测试', () => {
  it('create', () => {
    const { asFragment } = render(
      <Scrollbar height={60}>
        {Array(24).fill('item').map((item, index) => (
          <div key={index}>
            {item}
          </div>
        ))}
      </Scrollbar>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

