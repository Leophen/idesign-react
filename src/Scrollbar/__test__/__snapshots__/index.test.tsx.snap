// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scrollbar 组件测试 create 1`] = `
<DocumentFragment>
  <div
    class="i-scrollbar"
  >
    <div
      class="i-scrollbar__wrap"
      style="max-height: 60px; max-width: auto; user-select: unset;"
    >
      <div>
        item
      </div>
      <div>
        item
      </div>
      <div>
        item
      </div>
      <div>
        item
      </div>
      <div>
        item
      </div>
      <div>
        item
      </div>
      <div>
        item
      </div>
      <div>
        item
      </div>
      <div>
        item
      </div>
      <div>
        item
      </div>
      <div>
        item
      </div>
      <div>
        item
      </div>
      <div>
        item
      </div>
      <div>
        item
      </div>
      <div>
        item
      </div>
      <div>
        item
      </div>
      <div>
        item
      </div>
      <div>
        item
      </div>
      <div>
        item
      </div>
      <div>
        item
      </div>
      <div>
        item
      </div>
      <div>
        item
      </div>
      <div>
        item
      </div>
      <div>
        item
      </div>
    </div>
    <div
      class="i-scrollbar__barY"
    />
  </div>
</DocumentFragment>
`;
